src/Cli/Microsoft.DotNet.Cli.Utils/Uuid.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net;
+using System.Text;
+using System.Security.Cryptography;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class Uuid
+    {
+        /// <summary>
+        /// Generate a Version 5 (SHA1 Name Based) Guid from a name.
+        /// </summary>
+        /// <param name="name">The name to use for generating the GUID.</param>
+        /// <returns>A generated <see cref="GUID"/>.</returns>
+        public static Guid Create(string name)
+        {
+            // Any fixed GUID will do for a namespace.
+            Guid namespaceId = new Guid("28F1468D-672B-489A-8E0C-7C5B3030630C");
+
+            using (SHA1 hasher = SHA1.Create())
+            {
+                var nameBytes = System.Text.Encoding.UTF8.GetBytes(name ?? string.Empty);
+                var namespaceBytes = namespaceId.ToByteArray();
+
+                SwapGuidByteOrder(namespaceBytes);
+
+                var streamToHash = new byte[namespaceBytes.Length + nameBytes.Length];
+
+                Array.Copy(namespaceBytes, streamToHash, namespaceBytes.Length);
+                Array.Copy(nameBytes, 0, streamToHash, namespaceBytes.Length, nameBytes.Length);
+
+                var hashResult = hasher.ComputeHash(streamToHash);
+
+                var res = new byte[16];
+
+                Array.Copy(hashResult, res, res.Length);
+
+                unchecked { res[6] = (byte)(0x50 | (res[6] & 0x0F)); }
+                unchecked { res[8] = (byte)(0x40 | (res[8] & 0x3F)); }
+
+                SwapGuidByteOrder(res);
+
+                return new Guid(res);
+            }
+        }
+
+        /// <summary>
+        /// Generates a version 3 UUID given a namespace UUID and name. This is based on the algorithm described in
+        /// RFC 4122 (https://tools.ietf.org/html/rfc4122), section 4.3.
+        /// </summary>
+        /// <param name="namespaceUuid">The UUID representing the namespace.</param>
+        /// <param name="name">The name for which to generate a UUID within the given namespace.</param>
+        /// <returns>A UUID generated using the given namespace UUID and name.</returns>
+        public static Guid CreateUuid(Guid namespaceUuid, string name)
```

Comment:
This is code I wrote over 10 years ago for ASP.NET., but we can actually remove it. We use the v3 format to generate GUIDs in the Arcade tooling as it matches what WiX uses. I did use this to derive product codes in the CLI when I only have a workload pack ID to avoid some caching operations to obtain the msi JSON manifests.